### PR TITLE
MEPTS-655: New Indicator: Transferred In Patients

### PR DESCRIPTION
**Pull request by amosh101**

### CODE REVIEW

**Comment by cfaife** [comment]
the build is failing

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TransferredInCohortQueries.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    cd.addParameter(new Parameter("onOrAfter", "Start Date", Date.class));
+    cd.addParameter(new Parameter("onOrBefore", "End Date", Date.class));
+    cd.addParameter(new Parameter("location", "Location", Location.class));
+
+    CohortDefinition transferredIn = resumoMensalCohortQueries.getTransferredInPatients(false);
+
+    CohortDefinition txCurr = txCurrCohortQueries.getTxCurrCompositionCohort("txCurr", true);
+
+    String mappingsTrfIn = "onOrAfter=${onOrAfter},onOrBefore=${onOrBefore},location=${location}";
+    String mappingsCurr = "onOrBefore=${onOrBefore-3m},location=${location},locations=${location}";
```
Comment (cfaife) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TransferredInCohortQueries.java]: @amosh101  why do  we have 2 mapping for `location`?

Comment (amosh101) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TransferredInCohortQueries.java]: TxCurr takes both mappings as it is currently implemented. Fixing it was outside of scope of the ticket.

Comment (cfaife) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TransferredInCohortQueries.java]: I understand,  but this  `locations` mapping is only applied for the old spec and   txcurr  base cohort definition  which is  called in  DSD dataset.
image
Please have a look of the example in the  screenshot  above.

Comment (amosh101) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TransferredInCohortQueries.java]: Parameters simplified as recommended.

**Review by cfaife — APPROVED** [review]

**Pull request merged**